articles/200_coexistence_and_migration.md

```diff
@@ -0,0 +1,247 @@
+---
+layout: default
+title: Coexistence with and Migration from ROS 1
+permalink: articles/coexistence_and_migration.html
+abstract:
+  This article summarizes how the two major versions of ROS can coexist and be used together.
+  Additionally it covers the different aspects of migrating ROS packages from ROS 1 to ROS 2.
+author: '[Dirk Thomas](https://github.com/dirk-thomas)'
+published: true
+categories: Overview
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Goals
+
+### Separately usable
+
+* One version should not be affected by other versions choices / drawbacks.
+
+* ROS 1 users shouldn't be worried about ROS 2 development potentially negatively affecting ROS 1.
+
+  * No changes to existing ROS 1 packages should be necessary to continue to work.
+
+* ROS 2 users shouldn't need to make compromises due to ROS 1 choices / limitations / design decisions.
+
+### Support dual-homed applications
+
+To support creating dual-homed applications like the [ros1_bridge](https://github.com/ros2/ros1_bridge) ROS 1 and ROS 2 must be usable within a single application.
+
+### Minimize migration effort
+
+To ease the migration path the goal is to minimize the effort necessary where possible and when this goal doesn't conflict with the previously mentioned goals.
+
+## Technical Constraints
+
+### Namespace of ROS packages
+
+The first and very important question is about the namespace of ROS packages.
+Should ROS 1 and ROS 2 packages use separate namespaces or should they share the same namespace?
+For example can there be two packages names `foo` - one in ROS 1 and the other in ROS2 or do they have to be named differently.
+
+In the case that they use separate namespaces that does not prevent the release of a package `bar` into both, ROS 1 as well as ROS 2.
+
+Having them share a single namespace makes it difficult for users to distinguish them.
+
+Using separate namespaces allows e.g. to release a package `foo` in ROS 2 which provides the same API (but different implementation) as the ROS 1 package with the same name.
+This way downstream packages don't need to change when switching between the ROS 1 and ROS 2 version of the package.
+
+**Decision:** Use separate namespaces for packages in ROS 1 and ROS 2 (this is currently already the case).
```

Comment:
I think in this case we're referring to the debian package namespace. for example ros2 packages are released under `ros-r2b2-...` and ros1 package released under `ros-lunar-...`
This way if a package depends on `joy` and a user sourced a ROS2 workspace, he/she will get the headers from `ros-r2b2-joy`,  if he/she sourced a Lunar workspace he/she will get them from `ros-lunar-joy` without having to change the name of your dependencies or the code (if API is consistent)

Reply:
The term `namespace` isn't related to the Debian package names because that name is different from distro to distro.

If the ROS 1 and ROS 2 packages would share the same namespace that would imply that they are being listed in a single `distribution.yaml` file in the `rosdistro` repo.

Currently we have a separate `rosdistro` repo so they have "separate namespace". The same would be the case if the ROS 2  packages are always listed in a separate distro (like `r2b2`) even when they are stored in the same repo.

Reply:
:+1: right thanks for clarifying